extensions/lifecycle/tasks/LifecycleTask.js

```diff
@@ -210,49 +212,36 @@ class LifecycleTask extends BackbeatTask {
      * @param {array} deleteMarkers - delete markers list
      * @return {array} merge sorted array
      */
-    _mergeSortVersionsLists(versions, deleteMarkers) {
+    _mergeSortedVersionsAndDeleteMarkers(versions, deleteMarkers) {
         const sortedList = [];
         let vIdx = 0;
         let dmIdx = 0;
 
         while (vIdx < versions.length || dmIdx < deleteMarkers.length) {
             if (versions[vIdx] === undefined) {
+                // versions list is empty, just need to merge remaining DM's
                 sortedList.push(deleteMarkers[dmIdx++]);
             } else if (deleteMarkers[dmIdx] === undefined) {
+                // DM's list is empty, just need to merge remaining versions
                 sortedList.push(versions[vIdx++]);
             } else if (versions[vIdx].Key !== deleteMarkers[dmIdx].Key) {
-                // 1. by Key name
+                // 1. by Key name, alphabetical order sorted by ascii values
                 const isVKeyEarlier = (versions[vIdx].Key <
                     deleteMarkers[dmIdx].Key);
                 if (isVKeyEarlier) {
                     sortedList.push(versions[vIdx++]);
                 } else {
                     sortedList.push(deleteMarkers[dmIdx++]);
                 }
-            } else if (versions[vIdx].LastModified !==
-            deleteMarkers[dmIdx].LastModified) {
-                // 2. by LastModified
-                const isVDateEarlier = (new Date(versions[vIdx].LastModified) >
-                    new Date(deleteMarkers[dmIdx].LastModified));
-                if (isVDateEarlier) {
-                    sortedList.push(versions[vIdx++]);
-                } else {
-                    sortedList.push(deleteMarkers[dmIdx++]);
-                }
-            } else if (versions[vIdx].IsLatest) {
-                // 3. if both above match,
-                //    find if one `IsLatest` and take preference
-                sortedList.push(versions[vIdx++]);
-            } else if (deleteMarkers[dmIdx].IsLatest) {
-                sortedList.push(deleteMarkers[dmIdx++]);
             } else {
-                // 4. None of above? Check VersionId
-                const isVVersionIdGreater = (versions[vIdx].VersionId >
+                // 2. by VersionId, lower means newer
+                const isVersionVIdGreater = (versions[vIdx].VersionId >
```

Comment:
Similar here, although the var is quite explicit, for consistency it may be renamed to `isVersionVidOlder` (or reversed logic with `isVersionVidNewer` to keep the same logic than above) - which makes the comment at line 241 unnecessary.